scene/main/viewport.cpp

```diff
@@ -3252,6 +3268,12 @@ void Viewport::_mouse_leave_viewport() {
 }
 
 void Viewport::_drop_mouse_over(Control *p_until_control) {
+	if (gui.sending_mouse_enter_exit_notifications) {
+		// If notifications are already being sent, defer call.
+		callable_mp(this, &Viewport::_drop_mouse_over).call_deferred(p_until_control);
+		return;
+	}
```

Comment:
I'm not familiar with this code, but slightly worried at the possibility of a deferred infinite recursion here. Are we absolutely sure that this won't end up deferring itself over and over again without ever reaching a condition that would make `gui.sending_mouse_enter_exit_notifications` false?

Reply:
Yes, since every time `gui.sending_mouse_enter_exit_notifications` is set to true, it is immediately set to false in the same scope after it is done sending a fixed amount of notifications.
I'm not sure if there is a way to tell if this call was already deferred, but I could add a check just in case something changes and it is set to true?

Also this method only removes Controls from a list and cannot add any, so it cannot keep calling itself infinitely from user code, like was previously possible in `_gui_update_mouse_over`.